app.ts

```diff
@@ -0,0 +1,216 @@
+window.onload = () => {
+
+    let numColumns: number;
+    let columns: any;
+    let recordCount: number;
+    let startIndex = 0; // Default value = 0;
+    let tableRecordCount = 20; // Default value = 20;
+    let timer: number;
+
+    // Canvas:
+    let canvas = document.createElement("div");
+    canvas.id = "canvas";
+    $("body").append(canvas);
+
+    // Buttons and fields:
+    let searchbtn = document.createElement("button");
+    searchbtn.innerText = "Search";
+
+    let nextbtn = document.createElement("button");
+    nextbtn.innerText = "Next Page";
+
+    let prevbtn = document.createElement("button");
+    prevbtn.innerText = "Previous Page";
+
+    let fromField = document.createElement("input");
+    fromField.placeholder = "from";
+
+    // Window resizing (using a debouncing method):
+    window.onresize = () => {
```

Comment:
Also, please do not declare this callback while you're declaring the onload callback, rather move it out into a fresh scope. Declaring the other onclick handlers for the HTML elements are fine, but declaring this new callback on the same object feels wrong to me.